### ISSUE TITLE
cabal new-install haddock-api fails with GHC 8.8.2

### ISSUE THREAD

**Issue by alanz** [haskell/haddock#1123]
`cabal new-install haddock-api --with-compiler=ghc-8.8.2` fails with

```
src/Haddock/Convert.hs:621:39: error:
    • Variable not in scope: injectiveVarsOfBinder :: TyConBinder -> FV
    • Perhaps you meant ‘injectiveVarsOfType’ (imported from TyCoRep)
    |
621 |                            mapUnionFV injectiveVarsOfBinder dropped_binders
    |                                       ^^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build haddock-api-2.23.0. See the build log above for
details.
```
cabal index : `hackage.haskell.org,2020-01-24T20:23:40Z`

Building from https://github.com/haskell/haddock at be8b02c4e3cffe7d45b3dad0a0f071d35a274d65 succeeds, but pulls in `haddock-library-1.8.0.1`.

```
~/mysrc/github/haskell/haddock$ cabal new-install --with-compiler=ghc-8.8.2
Wrote tarball sdist to
/home/alanz/mysrc/github/haskell/haddock/dist-newstyle/sdist/haddock-2.23.0.tar.gz
Wrote tarball sdist to
/home/alanz/mysrc/github/haskell/haddock/dist-newstyle/sdist/haddock-api-2.23.0.tar.gz
Wrote tarball sdist to
/home/alanz/mysrc/github/haskell/haddock/dist-newstyle/sdist/haddock-library-1.8.0.1.tar.gz
Wrote tarball sdist to
/home/alanz/mysrc/github/haskell/haddock/dist-newstyle/sdist/haddock-test-0.0.1.tar.gz
Resolving dependencies...
Build profile: -w ghc-8.8.2 -O1
In order, the following will be built (use -v for more details):
 - haddock-library-1.8.0.1 (lib) (requires build)
 - haddock-api-2.23.0 (lib) (requires build)
 - haddock-2.23.0 (exe:haddock) (requires build)
Starting     haddock-library-1.8.0.1 (lib)
Building     haddock-library-1.8.0.1 (lib)
Installing   haddock-library-1.8.0.1 (lib)
Completed    haddock-library-1.8.0.1 (lib)
Starting     haddock-api-2.23.0 (lib)
Building     haddock-api-2.23.0 (lib)
Installing   haddock-api-2.23.0 (lib)
Completed    haddock-api-2.23.0 (lib)
Starting     haddock-2.23.0 (exe:haddock)
Building     haddock-2.23.0 (exe:haddock)
Installing   haddock-2.23.0 (exe:haddock)
Completed    haddock-2.23.0 (exe:haddock)
Symlinking 'haddock'
```
So it seems `haddock-library-1.8.0.1` needs to go onto hackage?

**Comment by jneira** [comment]
Any update on this? haskell-ide-engine users are hitting this issue and we aim to make a workaround if there is no release in the short term

**Comment by harpocrates** [comment]
This situation is frustrating indeed. It looks like GHC made some changes directly to the 8.8 branch of Haddock in https://github.com/haskell/haddock/commit/e6ca100973c496cd98da3385594fa9a81320f7cb, breaking the build for 8.8.1 in the process. What's worse is that 8.8.2 can only build with this commit. 😠 

I'll see what I can do to make some sort of small release and bounds updates to Hackage to deal with this issue.

**Comment by harpocrates** [comment]
Alright. This should be fixed now - sorry for the long delay.

  * `haddock-api-2.23.0` had its bounds revised to build only with 8.8.1
  * `haddock-api-2.23.1` was released to build only with 8.8.2 or higher (and with virtually no user-visible changes compared to `2.23.0`)
